Gems/Atom/RHI/DX12/Code/Source/RHI/CommandListBase.h

```diff
@@ -67,8 +116,12 @@ namespace AZ
 
             RHI::HardwareQueueClass m_hardwareQueueClass;
             Microsoft::WRL::ComPtr<ID3D12GraphicsCommandListX> m_commandList;
-            AZStd::vector<D3D12_RESOURCE_BARRIER> m_queuedBarriers;
+            AZStd::vector<BarrierOp> m_queuedBarriers;
             bool m_isRecording = false;
+            struct State
+            {
+                RHI::MultisampleState m_customSamplePositions;
+            } m_baseState;
```

Comment:
It is a little confusing to have a State struct but still has so many loose variables outside it. What makes m_customSamplePositions special that it gets it's own struct?